ledger-tool/src/main.rs

```diff
@@ -1908,12 +1908,12 @@ fn main() {
             }
         }
         ("verify", Some(arg_matches)) => {
-            let accounts_index_config = value_t!(arg_matches, "accounts_index_bins", usize)
-                .ok()
-                .map(|bins| AccountsIndexConfig { bins: Some(bins) });
-
+            let mut accounts_index_config = AccountsIndexConfig::default();
+            if let Some(bins) = value_t!(matches, "accounts_index_bins", usize).ok() {
+                accounts_index_config.bins = Some(bins);
+            }
             let accounts_db_config = Some(AccountsDbConfig {
-                index: accounts_index_config,
+                index: Some(accounts_index_config),
```

Comment:
Looks like now `AccountsDbConfig::index` is always `Some`. Is this going to change in future PRs? If not, would it worthwhile to remove the `Option<>` on the `index` field in `AccountsDbConfig`?

Reply:
the option to pass None overall is still a valid option. It is true that currently validator and ledger-tool (verify) will always pass Some(...), but tests or other code paths may still pass None.

Reply:
it is a rust hassle to figure out whether one of several optional args are present and if so make the overall struct a Some vs None. So, I chose to make it a Some(struct with Nones) where then optional args fill in the Nones with Some per arg. This code will likely get cleaned up and even consolidated at some point by someone because arg handling is messy.